feat(kit): add new `addServerScanDir` composable (#24001)
nuxt/nuxt: docs/3.api/5.kit/11.nitro.md

  }
})
```

## `addServerRoutesDir`

Add a directory to be scanned for server routes by Nitro.

### Type

```ts

> function function addServerRoutesDir (dirs: string | string[], opts: { prepend?: boolean }): void

Reviewer: ```suggestion
function addServerRoutesDir (dirs: string | string[], opts: { prepend?: boolean }): void
```

```

### Parameters

#### `dirs`

**Type**: `string | string[]`

**Required**: `true`

> A route or an array of directories to register to be scanned for API routes by Nitro

Reviewer 2: ```suggestion
A directory or an array of directories to register to be scanned for API routes by Nitro
```

> A directory or an array of directories to register to be scanned for API routes by Nitro

Reviewer 3: To avoid confusion with (unprefixed) nitro routes, it would be better to be named AddAPIRoutesDir / AddServerAPIRoutesDir

Reviewer 4: what about `addServerScanDir` ? I think nitro will scan everything just like the `server` directory, meaning `/api`, `/routes` and `/utils` are all scanned right ?

Reviewer 3: Yeas, considering how it is implemented now, this utility adds scan dir. Would be also nice at the same time think of a consistent name for this functionality as we were also planning to introduce a nuxt kit utility that adds layer-like dirs that have `pages`, etc.)

  }
})
```

## `addServerScanDir`

Add directories to be scanned by Nitro. It will check for:

- `api/` folder (which will create API routes)

> - `middleware/` folder (which will register server middleware)

Reviewer 3: Since we are not listing all dirs, would be better to only list most useful ones `api/` and `utils/` (maybe `/routes` too, I guess `public/` should work too!)

There are more like `modules/`, `tasks/`, `routes/`, etc which I wish we had a page in nitro we could link to. For `/server/middleware` I am trying to reduce encouragement of it's use and since it is new and specially targeting programmatic users, probably better to not list in more places.

Reviewer 2: Would be nice to add the list of scanner dirs in Nitro docs so we could link from it

Reviewer 3: PR welcome to add the page 👍🏼

Reviewer 2: Should it be a section on https://nitro.unjs.io/guide/auto-imports ?

Reviewer 3: I guess we can make a new `guide/directory-structure/` section for supported scan dirs similar to nuxt..

Author: Sure. Let's cut this down and add a link later.

- `utils/` folder (which will register auto-imports within your Nitro server)

### Type

```ts
function addServerScanDir (dirs: string | string[], opts: { prepend?: boolean }): void
```

### Parameters

#### `dirs`

**Type**: `string | string[]`

**Required**: `true`

> A directory or an array of directories to register to be scanned for API routes by Nitro

Reviewer 3: ```suggestion
A directory or an array of directories to register to be scanned for by Nitro as server dirs.
```
